src/OpenTelemetry/Metrics/HistogramBuckets.cs

```diff
@@ -16,26 +16,37 @@
 
 namespace OpenTelemetry.Metrics
 {
+    /// <summary>
+    /// A collection of <see cref="HistogramBucket"/>s associated with a histogram metric type.
+    /// </summary>
+    // Note: Does not implement IEnumerable<> to prevent accidental boxing.
     public class HistogramBuckets
     {
-        internal readonly long[] BucketCounts;
+        internal readonly double[] ExplicitBounds;
 
-        internal readonly long[] AggregatedBucketCounts;
+        internal readonly long[] RunningBucketCounts;
 
-        internal readonly double[] ExplicitBounds;
+        internal readonly long[] SnapshotBucketCounts;
 
-        internal readonly object LockObject;
+        internal double RunningSum;
 
-        internal HistogramBuckets(double[] histogramBounds)
+        internal double SnapshotSum;
+
+        internal HistogramBuckets(double[] explicitBounds)
         {
-            this.ExplicitBounds = histogramBounds;
-            this.BucketCounts = histogramBounds != null ? new long[histogramBounds.Length + 1] : null;
-            this.AggregatedBucketCounts = histogramBounds != null ? new long[histogramBounds.Length + 1] : null;
-            this.LockObject = new object();
+            this.ExplicitBounds = explicitBounds;
+            this.RunningBucketCounts = explicitBounds != null ? new long[explicitBounds.Length + 1] : null;
+            this.SnapshotBucketCounts = explicitBounds != null ? new long[explicitBounds.Length + 1] : new long[0];
```

Comment:
Use `Array.Empty<long>()`?

Reply:
I'm using `SnapshotBucketCounts` for the lock now so it has to be something uniquely held by each instance of `HistogramBuckets`. The `Array.Empty<T>` instances are shared (AFAIK) so that might lead to some contention.

Reply:
Why not lock on `this.HistogramBuckets` in `MetricPoint?

Reply:
We return the `HistogramBuckets` instance to users in the `MetricPoint.GetHistogramBuckets()` method. So they could (in theory) use that for their own locking purposes. That could lead to contention or (worse case) deadlock. I wanted to use something internal. Basically this concept of [weak identity vs strong identity](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2002).